qa/setup_packages.py

```diff
@@ -16,34 +16,433 @@
     # Fall back to Python 2's urllib2
     from urllib2 import urlopen, HTTPError, Request
 
-# keeps names of all required packages as a dict key
-# required versions are list or dict with keys of CUDA version, to use default just put None
-# instead of version number, direct link can be used
-# put {0} in pacage link as a placeholder for python pip package version (i.e. cp27-cp27mu-linux_x86_64)
-# and cuda_v for cuXX version
-# NOTE: First version will be picked in case of one_config_only
-
-packages = {
-            "opencv-python" : ["4.2.0.32"],
-            "cupy-cuda{cuda_v}" : {
-                        "90" : ["6.6.0"],
-                        "100" : ["6.6.0"]},
-            "mxnet-cu{cuda_v}" : {
-                        "90" : ["1.6.0"],
-                        "100" : ["1.5.1"]},
-            "tensorflow-gpu" : {
-                        "90": ["1.12.0",],
-                        "100": ["1.14.0", "1.15.2", "2.0.1", "2.1.0"]},
-            "torch" : {
-                        "90": ["http://download.pytorch.org/whl/{cuda_v}/torch-1.1.0-{0}.whl"],
-                        "100": ["http://download.pytorch.org/whl/{cuda_v}/torch-1.4.0+{cuda_v}-{0}.whl"]},
-            "torchvision" : {
-                        "90": ["https://download.pytorch.org/whl/{cuda_v}/torchvision-0.3.0-{0}.whl"],
-                        "100": ["https://download.pytorch.org/whl/{cuda_v}/torchvision-0.5.0+{cuda_v}-{0}.whl"]},
-            "paddle" : {
-                        "90": ["https://paddle-wheel.bj.bcebos.com/gcc54/latest-gpu-cuda9-cudnn7-openblas/paddlepaddle_gpu-latest-{0}.whl"],
-                        "100": ["https://paddle-wheel.bj.bcebos.com/gcc54/latest-gpu-cuda10-cudnn7-openblas/paddlepaddle_gpu-latest-{0}.whl"]},
-            }
+PYTHON_VERSION = ".".join([str(x) for x in sys.version_info[0:2]])
+
+class PckgVer():
+    """Class that holds a version string accompanied with maximum and minimum python version that
+        this version should support. If python falls beyond version bounds it evaluates to the empty string
+
+        Parameters
+        ----------
+        `ver`: str
+            Version that is housed by object of this class
+        `python_max_ver` : str, optional, default = None
+            Maximum python version supported by this package. If empty there is no upper bound
+        `python_min_ver`: str, optional, default = None
+            Mimimum python version supported by this package. If empty there is no lower bound
+    """
+    def __init__(self, ver, python_max_ver=None, python_min_ver=None):
```

Comment:
You have a weird order of min and max for an "interval" :P

Reply:
I though that max version would be more frequently used so you can write just `PckgVer("2.0.1",  "3.7")` instead of `PckgVer("2.0.1",  python_max_ver="3.7")`.
But I agree it is not always intuitive. Fixed.